engine/execution/computation/computer/uploader/uploader_test.go

```diff
@@ -0,0 +1,116 @@
+package uploader
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"sync"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+func Test_AsyncUploader(t *testing.T) {
+
+	wgCalled := sync.WaitGroup{}
+	wgCalled.Add(3)
+
+	wgAllDone := sync.WaitGroup{}
+	wgAllDone.Add(1)
+
+	uploader := &DummyUploader{
+		f: func() {
+			// this should be called 3 times
+			wgCalled.Done()
+
+			wgAllDone.Wait()
+		},
+	}
+
+	metrics := &DummyCollector{}
+	async := NewAsyncUploader(uploader, zerolog.Nop(), metrics)
+
+	err := async.Upload(nil)
+	require.NoError(t, err)
+
+	err = async.Upload(nil)
+	require.NoError(t, err)
+
+	err = async.Upload(nil)
+	require.NoError(t, err)
+
+	wgCalled.Wait() // all three are in progress, check metrics
+
+	require.Equal(t, int64(3), metrics.Counter.Load())
+
+	wgAllDone.Done() //release all
+
+	<-async.Done()
+
+	require.Equal(t, int64(0), metrics.Counter.Load())
+}
+
+func Test_GCPBucketUploader(t *testing.T) {
+
+	t.Skip("requires GCP Bucket setup")
+
+	bucketName := os.Getenv("FLOW_TEST_GCP_BUCKET")
```

Comment:
What are the expectations on running this test? In particular, do I need a signed URL to run this?

Reply:
You need to have configured access to a named GCP bucket